### PR TITLE
[WIP] Add purchase_amendment

### PR DESCRIPTION
**Pull request by guewen**
**Work in progress**

Depends on
- [ ] https://github.com/odoo/odoo/pull/6038
- [ ] OCA/OCB#208
- [ ] https://github.com/OCA/stock-logistics-workflow/pull/94

### CODE REVIEW

File: purchase_amendment/README.rst [hunk 1]
```diff
@@ -0,0 +1,52 @@
+Purchase Amendment
+==============
```
Comment (pedrobaeza) [inline purchase_amendment/README.rst]: same length of title

File: purchase_amendment/README.rst [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+This module is meant to assist the users in handling the cases when a customer
+changes his mind about a purchase.
+
+See also `sale_amendment` (in https://github.com/OCA/sale-workflow)
+which cover a similar use case for changes in sale orders.
```
Comment (pedrobaeza) [inline purchase_amendment/README.rst]: s/cover/covers

File: purchase_amendment/README.rst [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+
+Background information
```
Comment (pedrobaeza) [inline purchase_amendment/README.rst]: This section can be dropped

File: purchase_amendment/model/purchase_order_amendment.py [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+        purchase = self.env['purchase.order'].browse(purchase_ids)
+
+        if any(inv.state != 'cancel' for inv in purchase.invoice_ids):
+            raise exceptions.Warning(
+                _('An invoiced order cannot be amended.')
+            )
+
+        items = []
+        for line in purchase.order_line:
+            if line.state in ('cancel', 'done'):
```
Comment (pedrobaeza) [inline purchase_amendment/model/purchase_order_amendment.py:60]: `if line.state in ('cancel', 'done') or line.amended_by_ids:`

**Comment by gurneyalex** [comment]
:+1:

**Comment by lepistone** [comment]
superseded by #138

**Pull request closed without merging**